transport/ws/src/main/java/org/kaazing/gateway/transport/ws/extension/WebSocketExtensionFactory.java

```diff
@@ -19,36 +19,63 @@
 import static java.util.Collections.unmodifiableMap;
 import static java.util.ServiceLoader.load;
 
-import java.io.IOException;
 import java.net.ProtocolException;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 import java.util.ServiceLoader;
 
 import org.kaazing.gateway.resource.address.ws.WsResourceAddress;
 import org.kaazing.gateway.transport.http.HttpAcceptSession;
-import org.kaazing.gateway.transport.ws.extension.ExtensionHeader.EndpointKind;
+import org.kaazing.gateway.transport.ws.extension.WebSocketExtensionFactorySpi.ExtensionOrderCategory;
 
 public final class WebSocketExtensionFactory {
 
     private final Map<String, WebSocketExtensionFactorySpi> factoriesRO;
     private final List<ExtensionHeader> supportedExtensionHeaders;
+    // note: extension names are in the order they would like to be on the pipeline in.
```

Comment:
What does "on the pipeline in" mean? For clarity I suggest rewording to something like the following:
extension names are in the order they would like to be on the pipeline when a frame is being written out to the network